manifest/pack_pull_git.go

```diff
@@ -6,18 +6,99 @@ package manifest
 
 import (
 	"context"
-	"encoding/hex"
 	"fmt"
+	"io"
+	"math"
+	"math/rand"
 	"net/url"
 	"strings"
+	"sync"
+	"time"
 
-	libgit "github.com/libgit2/git2go/v31"
+	"github.com/go-git/go-git/v5"
+	gitplumbing "github.com/go-git/go-git/v5/plumbing"
+	githttp "github.com/go-git/go-git/v5/plumbing/transport/http"
+	gitssh "github.com/go-git/go-git/v5/plumbing/transport/ssh"
 
 	"kraftkit.sh/log"
 	"kraftkit.sh/pack"
 	"kraftkit.sh/unikraft"
 )
 
+type cloneProgress struct {
+	io.Writer
+
+	once           sync.Once
+	onProgress     func(progress float64)
+	completeParent chan bool
+	completeWorker chan bool
+}
+
+func (p cloneProgress) Write(b []byte) (int, error) {
+	if p.onProgress == nil {
+		return len(b), nil
+	}
+
+	var prog string
+	var scale float64
+
+	// Trim prefix
+	if strings.HasPrefix(string(b), "Counting objects") {
+		prog = strings.TrimPrefix(string(b), "Counting objects: ")
+		scale = 0.1
+	} else if strings.HasPrefix(string(b), "Compressing objects") {
+		prog = strings.TrimPrefix(string(b), "Compressing objects: ")
+		scale = 0.2
+	} else if strings.HasPrefix(string(b), "Total ") {
```

Comment:
nit(style): a good candidate for a `switch`

```suggestion
	switch {
	case strings.HasPrefix(string(b), "Counting objects"):
		prog = strings.TrimPrefix(string(b), "Counting objects: ")
		scale = 0.1
	case strings.HasPrefix(string(b), "Compressing objects"):
		prog = strings.TrimPrefix(string(b), "Compressing objects: ")
		scale = 0.2
	case strings.HasPrefix(string(b), "Total "):
```